### ISSUE TITLE
Use semantic Tailwind color names so the admin can be themed

### ISSUE THREAD

**Issue by tvdeyen** [solidusio/solidus#5848]
**Desired Behavior**

I would like to be able to theme the Solidus admin as I am currently able now with the legacy backend. There we have bootstrap variables that can be changed to new color values. The use of generic variables (`primary`, `secondary`, `danger`, `brand`, etc.) makes it very easy to change the look of Solidus admin. Also light and darkmode are easier to maintain that way.

**Current Behavior**

We currently have [color based class names](https://github.com/solidusio/solidus/blob/e8c9675c4f15792b2a039a340232834bb9e24037/admin/config/tailwind.config.js#L17) all over the new admin. This makes it very hard to change the look of the new admin. Also it makes it much harder for us to change the theme in the future and make adjustments to the light and dark modes.

**Additional notes**

Solidus is a framework that allows clients to adopt the admin to their needs. This includes theming. IMO this is one of reasons for the success of Solidus in the community. We should not remove this feature that many stores rely on.

In the tailwind docs are ways described to make this more maintainable for us. https://tailwindcss.com/docs/customizing-colors#naming-your-colors

I think we should move to a more framework-alike approach with the new admin.

**Comment by kennyadsl** [comment]
@rainerdema @elia, do you remember why we used this approach? Did we ever think about prioritizing theming before?

**Comment by elia** [comment]
@kennyadsl @tvdeyen yes, we just stuck with the TailswindCSS recommendation, additionally it was no time to devise a theming system, which is essentially another API to design, maintain, and migrate.

Sounds like we might have a really good reason to steer in that direction now, as the documentation says:

>Again, we recommend sticking to the default naming convention for most projects, and only using abstract names if you have a really good reason.

That said nailing the right level of abstraction with those colors is really hard, I tried it many times, and it will make interpreting tailwind classes harder (`bg-primary-intense` is not as clear as `bg-gray-900`) and would mean thinking about the role of each color, having multiple roles target the same colors and a bunch of other puzzles that are not straightforward to solve.

So I propose to limit the customization to something like the accent color, and go vanilla with the general design, I might be wrong, but that covers the need to match the style of a client in most cases. 

@tvdeyen would that cover your use-case? How deep you think the customization would need to be?

**Comment by tvdeyen** [comment]
> @kennyadsl @tvdeyen yes, we just stuck with the TailswindCSS recommendation, additionally it was no time to devise a theming system, which is essentially another API to design, maintain, and migrate.
> 
> Sounds like we might have a really good reason to steer in that direction now, as the documentation says:
> 
> > Again, we recommend sticking to the default naming convention for most projects, and only using abstract names if you have a really good reason.
> 
> That said nailing the right level of abstraction with those colors is really hard, I tried it many times, and it will make interpreting tailwind classes harder (`bg-primary-intense` is not as clear as `bg-gray-900`) and would mean thinking about the role of each color, having multiple roles target the same colors and a bunch of other puzzles that are not straightforward to solve.
> 
> So I propose to limit the customization to something like the accent color, and go vanilla with the general design, I might be wrong, but that covers the need to match the style of a client in most cases.

I agree. Limiting it to things like brand-color and maybe some state colors (danger/error, info, success/notice, warning/alert) should be enough to satisfy customization needs of stores.

> @tvdeyen would that cover your use-case? How deep you think the customization would need to be?

Yes, that will cover it and I think the customization level should be as shallow as possible. Just cover the basic needs.

One idea would be to read a `solidus/tailwind.config.js` file from stores (and potentially extensions) with the default Tailwind config from admin exported as module.

**Comment by fthobe** [comment]
Are branding features a topic only for the backend or is it something that you want to consider for starter front end as well?

**Comment by tvdeyen** [comment]
> Are branding features a topic only for the backend or is it something that you want to consider for starter front end as well?

Since the starter frontend generates a dedicated Rails app for each store, branding happens in that Rails app, not in Solidus.

**Comment by fthobe** [comment]
Seems fair.

**Comment by fthobe** [comment]
@tvdeyen 
> One idea would be to read a solidus/tailwind.config.js file from stores (and potentially extensions) with the default Tailwind config from admin exported as module.

What happens if people use it headless in your scenario? If I think about it the dependency on any preconfigured tailwind config is totally contrary on the whole "use solidus with your own frontend" doctrine. Wouldn't three backend configuration options like color 1, color 2 and logo cover pretty much everything? In addition given that the whole backend is pretty much grey apart from the logo, what do you want to style?
